### PR TITLE
data driven split serial implementations

### PR DESCRIPTION
**Pull request by evils**
the additions i made to get my keyboard working

## Description

additions to keyboard.jsonschema (with implementations in info_config.json and info_rules.json) for:  
split hand config
serial and serial driver config
via enable config
an a SPLIT_LAYER_STATE_ENABLE mapping i'm not sure i needed, but wrote...

## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [x] Enhancement/optimization
- [ ] Keyboard (addition or update)
- [ ] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Issues Fixed or Closed by This PR

* 

## Checklist

- [ ] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [x] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [ ] My change requires a change to the documentation.
  - not sure, i found finding the actual implementation the best documentation on using this feature...
- [ ] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

File: data/schemas/keyboard.jsonschema [hunk 1]
```diff
@@ -504,5 +513,46 @@
...
+                },
+            }
+        },
+        "via": {
+            "type": "object",
+            "additionalProperties": false,
+            "properties": {
+                "enabled": {"type": "boolean"},
+            }
+        },
```
Comment (tzarc) [inline data/schemas/keyboard.jsonschema:556]: in theory this should already be covered by:
```json
{
    "features": {
        "via": true
    }
}```

Comment (evils) [reply data/schemas/keyboard.jsonschema:556]: where is this from?
i can't find that in develop, the only occurrence there of `"via": true` is in a single keyboard's keymap.json

Comment (zvecr) [reply data/schemas/keyboard.jsonschema:556]: <https://github.com/qmk/qmk_firmware/blob/master/lib/python/qmk/info.py#L122>

File: data/schemas/keyboard.jsonschema [hunk 2]
```diff
@@ -439,6 +440,14 @@
                         "timeout": {"$ref": "qmk.definitions.v1#/unsigned_int"}
                     }
                 }
+                "hand": {
+                    "type": "object",
+                    "additionalProperties": false,
+                    "properties": {
+                        "pin": {"$ref": "qmk.definitions.v1#/mcu_pin"},
```
Comment (evils) [inline data/schemas/keyboard.jsonschema:447]: this option (SPLIT_HAND_PIN) also appears to be covered by info.py, as `split.main`
but that doesn't seem to work for me, the schema definition does have a split.main,
which has a `pin` option, but no way to set which pin to use

the associated low_is_left option seems to be missing from any merged implementations

File: data/schemas/keyboard.jsonschema (hunk 1 above)
Comment (evils) [reply data/schemas/keyboard.jsonschema:556]: jup, that works

File: data/schemas/keyboard.jsonschema (hunk 2 above)
Comment (sigprof) [reply data/schemas/keyboard.jsonschema:447]: Looks like [`_extract_split_main` in `lib/python/qmk/info.py`](https://github.com/qmk/qmk_firmware/blob/6e1c2b20e55517cc70a0bfb37523ae7dbe26a7d8/lib/python/qmk/info.py#L296-L344) must be fixed then (it extracts the pin info from `SPLIT_HAND_MATRIX_GRID`, but not from `SPLIT_HAND_PIN`). The problem is that if we take the existing `SPLIT_HAND_MATRIX_GRID` → `split.matrix_grid` conversion as an example, the new property would be named `split.pin`, and I'm not sure that would be understandable.

**Comment by evils** [comment]
resolved merge conflict

@zvecr does your self-assignment to this mean you have a plan for the changes?

**Comment by evils** [comment]
@zvecr it's been 3 weeks since you assigned yourself to this, do you have a plan for the changes
or should i just cut out the SPLIT_HAND_PIN part, which seems to be the blocking problem?

**Comment by evils** [comment]
@zvecr it's been 44 days since you assigned yourself to this, was that by mistake?

i've resolved another merge conflict

File: data/schemas/keyboard.jsonschema (hunk 2 above)
Comment (evils) [reply data/schemas/keyboard.jsonschema:530]: haven't looked further at this, but it seems like my approach is the clearer one, mostly by merit of existing...

would this PR be merged if i pull out the SPLIT_HAND_PIN part?

Comment (Kriechi) [reply data/schemas/keyboard.jsonschema:530]: for split handedness, there is a larger PR in the works: https://github.com/qmk/qmk_firmware/pull/18254

**Pull request closed without merging**